bundles/org.openhab.core/src/main/java/org/openhab/core/internal/service/WatchServiceImpl.java

```diff
@@ -248,17 +249,17 @@ private void notifyListeners(Path path) {
         Kind firstElement = kinds.get(0);
         Kind lastElement = kinds.get(kinds.size() - 1);
 
-        // determine final event
-        if (lastElement == Kind.DELETE) {
-            if (firstElement == Kind.CREATE) {
-                logger.debug("Discarding events for '{}' because file was immediately deleted bafter creation", path);
-                return;
-            }
-            doNotify(path, Kind.DELETE);
-        } else if (firstElement == Kind.CREATE) {
-            doNotify(path, Kind.CREATE);
-        } else {
-            doNotify(path, Kind.MODIFY);
+        if (firstElement == Kind.CREATE && lastElement == Kind.DELETE) {
+            logger.debug("Discarding events for '{}' because file was immediately deleted after creation", path);
+            return;
+        }
+
+        if (firstElement == Kind.CREATE || firstElement == Kind.DELETE) {
+            doNotify(path, firstElement);
+        }
+
+        if (kinds.size() > 1 && firstElement != lastElement) {
+            doNotify(path, lastElement);
```

Comment:
That's not correct. CREATE - MODIFY should result in only one event CREATE. There is no need for a second reload.

I still don't understand what real-life use-case can cause an issue with the way it is implemented now.

Reply:
I've updated the code again, and also added a test to clarify the input vs output sequence. CREATE - MODIFY should now only result in one event CREATE.

> I still don't understand what real-life use-case can cause an issue with the way it is implemented now.

MODIFY - DELETE - CREATE - DELETE - MODIFY (or a combination thereof) should send DELETE - CREATE so the listeners are aware that the files were deleted and recreated. This should still make existing listeners / handlers to work because they will still reload the file once, so it still solves the multiple reload problem, but at the same time improves the visibility of what truly happened.